[Feature][Connector V2] expose configurable options in Hudi

cason0126 opened the pull request:
subtask of https://github.com/apache/incubator-seatunnel/issues/3337

cason0126 committed:
add expose configurable options in Hudi

cason0126 committed:
fix codeStyle
[Feature][Connector V2]expose configurable options in Hudi

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java:
```diff
@@ -17,18 +17,44 @@
+    public static final Option<String> USE_KERBEROS =
+            Options.key("use.kerberos")
+                    .stringType()
+                    .noDefaultValue()
```
From the document I see `use.Kerberos` have a default value `false`.

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public OptionRule optionRule() {
+        return null;
```
Please return the OptionRule of this connector.

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
fix use.kerberos type from string to boolean
fix use.kerberos default with false

cason0126 committed:
Merge branch 'apache:dev' into add_hudi_option

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder().required(TABLE_PATH,TABLE_TYPE,CONF_FILES).build();
```
You lost the option options `use.kerberos` `kerberos.principal` `kerberos.principal.file`. And `kerberos.principal` `kerberos.principal.file` is required when `use.kerberos=true`. So you need add `conditional` . You can reference https://github.com/apache/incubator-seatunnel/pull/3320/files#diff-0626e040d3b207c066aee2e5e40f7faaf3d6dc1d01a87d097a21136b7b7092e3

cason0126 replied:
OK，I will FIX THEM SOON

cason0126 replied:
@EricJoy2048

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
fix optionRule;
Update Hudi Connector Doc About Config;

cason0126 committed:
Merge remote-tracking branch 'origin/add_hudi_option' into add_hudi_option

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
fix  table.type Description

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
fix  codeStyle;
Avoid Using .*;

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/config/HudiSourceConfig.java:
```diff
@@ -17,18 +17,44 @@
+    public static final Option<Boolean> USE_KERBEROS =
+            Options.key("use.kerberos")
+                    .booleanType()
+                    .defaultValue(false)
```
Should  we make `use_kerberos`  as an enum type option?

EricJoy2048 replied:
> Should we make `use_kerberos` as an enum type option?

Sorry, You don't need update it to enum type.

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:
```diff
@@ -0,0 +1,44 @@
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(TABLE_PATH,TABLE_TYPE,CONF_FILES)
+                .optional(USE_KERBEROS,KERBEROS_PRINCIPAL,KERBEROS_PRINCIPAL_FILE)
```
```suggestion
                .optional(USE_KERBEROS)
```

cason0126 replied:
DONE

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
Remove Option : KERBEROS_PRINCIPAL,KERBEROS_PRINCIPAL_FILE From
 optional In optionRule

EricJoy2048 commented:
Please fix code style check

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
fix codeStyle

cason0126 commented:
I have fix code sytle check ; 
wating workflows approval ;
@EricJoy2048  THX!

EricJoy2048 commented on seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSourceFactory.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Override
+    public OptionRule optionRule() {
```
Add UT for `optionRule()` please. We add some check rules for OptionRule builder, but need you add an UT to trigger it.

cason0126 replied:
OK, I Will Add it Soon .

cason0126 replied:
DONE

cason0126 committed:
[Feature][Connector V2]expose configurable options in Hudi
Add UT for Hudi option rule

EricJoy2048 commented:
Please resolve conflicts.

cason0126 committed:
Merge branch 'dev' into add_hudi_option

# Conflicts:
#	seatunnel-connectors-v2/connector-hudi/src/main/java/org/apache/seatunnel/connectors/seatunnel/hudi/source/HudiSource.java

cason0126 committed:
[Feature][Connector V2]expose
Resolve Conflicts and fix comment text

EricJoy2048 commented:
fix code style please.

cason0126 committed:
[Feature][Connector V2] Hudi Fix CodeStyle

cason0126 commented:
OK。Sorry about that.
CS Setting does not including test source . 
I have fix it .

cason0126 commented:
How can I fix this problem ：Dead Links Fail 。
I do nothing but oneline to fix codestyle。 then，it happens。
@EricJoy2048

cason0126 committed:
Merge branch 'dev' into add_hudi_option

cason0126 commented:
fix dead link

EricJoy2048 commented:
You need merge dev branch code to fix the dead link.

cason0126 committed:
Merge branch 'dev' into add_hudi_option

cason0126 commented:
is there anything to fix?
@EricJoy2048

EricJoy2048 reviewed (approved):
LGTM

liugddx reviewed (approved):
+1
